### PR TITLE
Slider shows visual label of value on focus

### LINKED ISSUE #113538: Slider does not show value indicator when focused.
### Description

When working on other `Slider` issues, I noticed discrete `Slider` does not show the value indicator when focused.

<details> 
<summary>code sample</summary> 

```dart
import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      themeMode: ThemeMode.light,
      theme: ThemeData(
        useMaterial3: true,
        brightness: Brightness.light,
        colorSchemeSeed: const Color(0xff6750a4),
      ),
      darkTheme: ThemeData(
        useMaterial3: true,
        brightness: Brightness.dark,
        colorSchemeSeed: const Color(0xff6750a4),
      ),
      home: const SliderExample(),
    );
  }
}

class SliderExample extends StatefulWidget {
  const SliderExample({super.key});

  @override
  State<SliderExample> createState() => _SliderExampleState();
}

class _SliderExampleState extends State<SliderExample> {
  double sliderValue = 40;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Slider'),
      ),
      body: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: <Widget>[
          const Text('Continuous'),
          Slider(
            value: sliderValue,
            max: 100,
            label: sliderValue.round().toString(),
            onChanged: (double value) {
              setState(() {
                sliderValue = value;
              });
            },
          ),
          const SizedBox(height: 10),
          const Text('Discrete'),
          Slider(
            value: sliderValue,
            max: 100,
            divisions: 5,
            label: sliderValue.round().toString(),
            onChanged: (double value) {
              setState(() {
                sliderValue = value;
              });
            },
          ),
        ],
      ),
    );
  }
}

``` 
	
</details>

### Material web slider demo
ezgif com-gif-maker

### Flutter
 
ezgif com-gif-maker (1)

(This is compared to https://material-components.github.io/material-web/demos/slider/)

Perhaps this is a regression.

### PR DESCRIPTION
**Pull request by DBowen33**
Fixed issue to where value indicator label is not shown to user when initially focused.

Before:  [Screen recording 2024-08-05 12.16.24 PM.webm](https://github.com/user-attachments/assets/806e94d7-7c6c-40e5-93af-dbf2fb9e0dd4)
After: https://screencast.googleplex.com/cast/NTY5NzIzMTYxNjIxMjk5MnxkY2IyMGNkYi1iZA

Fixes https://github.com/flutter/flutter/issues/113538

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

**Comment by Piinks** [comment]
FYI @TahaTesser in case this affects the slider rewrite (which I need to catch up on reviewing!)

File: packages/flutter/test/material/slider_test.dart [hunk 1]
```diff
@@ -4548,4 +4548,60 @@ void main() {
...
+    final FocusNode focusNode = FocusNode();
+    addTearDown(focusNode.dispose);
+
+    Widget buildApp({bool enabled = true}) {
+      return MaterialApp(
+        theme: ThemeData(
+          sliderTheme: const SliderThemeData(
+            showValueIndicator: ShowValueIndicator.always,
+          ),
+        ),
```
Comment (QuncCccccc) [inline packages/flutter/test/material/slider_test.dart:4563]: nit: Seems we can remove this part?

File: packages/flutter/test/material/slider_test.dart [hunk 2]
```diff
@@ -4548,4 +4548,60 @@ void main() {
       paints..scale()..path(color: theme.colorScheme.primary),
     );
   }, variant: TargetPlatformVariant.desktop());
+
+  testWidgets('Value indicator label is shown when focused', (WidgetTester tester) async {
+    double value = 0.5;
+    final FocusNode focusNode = FocusNode();
+    addTearDown(focusNode.dispose);
+
+    Widget buildApp({bool enabled = true}) {
```
Comment (QuncCccccc) [inline packages/flutter/test/material/slider_test.dart:4557]: nit: maybe we can remove the parameter here since we only tested the enable slider:)

**Review by QuncCccccc — APPROVED** [review]
LGTM:) Thanks for the fix!

**Review by MitchellGoodwin — APPROVED** [review]
LGTM!

File: packages/flutter/test/material/slider_test.dart (hunk 1 above)
Comment (DBowen33) [reply packages/flutter/test/material/slider_test.dart]: Done!

File: packages/flutter/test/material/slider_test.dart (hunk 2 above)
Comment (DBowen33) [reply packages/flutter/test/material/slider_test.dart]: Done, thanks for the recs!

**Pull request merged**